Failed to build qtmultimedia for target i686-w64-mingw32.shared.dw2

Trying to build qtwebkit on a fresh mxe clone and stumbled over this along the way.

```
make --jobs=4 qtwebkit
...

[build]       qtbase                 i686-w64-mingw32.shared.dw2
[done]        qtbase                 i686-w64-mingw32.shared.dw2                          654948 KiB     55m3.915s
[build]       qtimageformats         i686-w64-mingw32.shared.dw2
[build]       qtsvg                  i686-w64-mingw32.shared.dw2
[build]       qtxmlpatterns          i686-w64-mingw32.shared.dw2
[build]       qtsensors              i686-w64-mingw32.shared.dw2
[done]        qtimageformats         i686-w64-mingw32.shared.dw2                          12452 KiB      2m26.069s
[build]       qtwebsockets           i686-w64-mingw32.shared.dw2
[done]        qtsvg                  i686-w64-mingw32.shared.dw2                          18192 KiB      2m42.086s
[done]        qtsensors              i686-w64-mingw32.shared.dw2                          14592 KiB      4m8.261s
[done]        qtwebsockets           i686-w64-mingw32.shared.dw2                          5460 KiB       2m7.878s
[done]        qtxmlpatterns          i686-w64-mingw32.shared.dw2                          333740 KiB     12m25.598s
[build]       qtdeclarative          i686-w64-mingw32.shared.dw2
[done]        qtdeclarative          i686-w64-mingw32.shared.dw2                          410064 KiB     34m25.650s
[build]       qtmultimedia           i686-w64-mingw32.shared.dw2
[build]       qtquickcontrols        i686-w64-mingw32.shared.dw2
[build]       qtwebchannel           i686-w64-mingw32.shared.dw2
[done]        qtwebchannel           i686-w64-mingw32.shared.dw2                          4884 KiB       1m17.309s
[done]        qtquickcontrols        i686-w64-mingw32.shared.dw2                          31412 KiB      8m27.569s

Failed to build package qtmultimedia for target i686-w64-mingw32.shared.dw2!
------------------------------------------------------------
i686-w64-mingw32.shared.dw2-g++ -c -pipe -fno-keep-inline-dllexport -O2 -std=c++1y -fno-exceptions -Wall -Wextra -Wextra -Wvla -Wdate-time -DUNICODE -D_UNICODE -DWIN32 -DMINGW_HAS_SECURE_API=1 -DQT_NO_LINKED_LIST -DQT_NO_FOREACH -DQT_NO_JAVA_STYLE_ITERATORS -DQT_NO_LINKED_LIST -DNO_DSHOW_STRSAFE -D_CRT_SECURE_NO_WARNINGS -DQT_DEPRECATED_WARNINGS -DQT_NO_NARROWING_CONVERSIONS_IN_CONNECT -DQT_NO_EXCEPTIONS -DQT_NO_DEBUG -DQT_PLUGIN -DQT_MULTIMEDIA_LIB -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -DQ_USE_SYBASE -I. -Icommon -Iplayer -I../common/evr -Icamera -I../../../include/QtMultimedia/5.15.2 -I../../../include/QtMultimedia/5.15.2/QtMultimedia -I../../../include -I../../../include/QtMultimedia -I/Users/holgermacmini/dev/mxe_qt5152/usr/i686-w64-mingw32.shared.dw2/qt5/include/QtGui/5.15.2 -I/Users/holgermacmini/dev/mxe_qt5152/usr/i686-w64-mingw32.shared.dw2/qt5/include/QtGui/5.15.2/QtGui -I/Users/holgermacmini/dev/mxe_qt5152/usr/i686-w64-mingw32.shared.dw2/qt5/include -I/Users/holgermacmini/dev/mxe_qt5152/usr/i686-w64-mingw32.shared.dw2/qt5/include/QtWidgets -I/Users/holgermacmini/dev/mxe_qt5152/usr/i686-w64-mingw32.shared.dw2/qt5/include/QtGui -I/Users/holgermacmini/dev/mxe_qt5152/usr/i686-w64-mingw32.shared.dw2/qt5/include/QtCore/5.15.2 -I/Users/holgermacmini/dev/mxe_qt5152/usr/i686-w64-mingw32.shared.dw2/qt5/include/QtCore/5.15.2/QtCore -I/Users/holgermacmini/dev/mxe_qt5152/usr/i686-w64-mingw32.shared.dw2/qt5/include/QtNetwork -I/Users/holgermacmini/dev/mxe_qt5152/usr/i686-w64-mingw32.shared.dw2/qt5/include/QtCore -I.moc/release -I/Users/holgermacmini/dev/mxe_qt5152/usr/i686-w64-mingw32.shared.dw2/qt5/mkspecs/win32-g++  -o .obj/release/moc_directshowcamerazoomcontrol.o .moc/release/moc_directshowcamerazoomcontrol.cpp
i686-w64-mingw32.shared.dw2-g++ -c -pipe -fno-keep-inline-dllexport -O2 -std=c++1y -fno-exceptions -Wall -Wextra -Wextra -Wvla -Wdate-time -DUNICODE -D_UNICODE -DWIN32 -DMINGW_HAS_SECURE_API=1 -DQT_NO_LINKED_LIST -DQT_NO_FOREACH -DQT_NO_JAVA_STYLE_ITERATORS -DQT_NO_LINKED_LIST -DNO_DSHOW_STRSAFE -D_CRT_SECURE_NO_WARNINGS -DQT_DEPRECATED_WARNINGS -DQT_NO_NARROWING_CONVERSIONS_IN_CONNECT -DQT_NO_EXCEPTIONS -DQT_NO_DEBUG -DQT_PLUGIN -DQT_MULTIMEDIA_LIB -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -DQ_USE_SYBASE -I. -Icommon -Iplayer -I../common/evr -Icamera -I../../../include/QtMultimedia/5.15.2 -I../../../include/QtMultimedia/5.15.2/QtMultimedia -I../../../include -I../../../include/QtMultimedia -I/Users/holgermacmini/dev/mxe_qt5152/usr/i686-w64-mingw32.shared.dw2/qt5/include/QtGui/5.15.2 -I/Users/holgermacmini/dev/mxe_qt5152/usr/i686-w64-mingw32.shared.dw2/qt5/include/QtGui/5.15.2/QtGui -I/Users/holgermacmini/dev/mxe_qt5152/usr/i686-w64-mingw32.shared.dw2/qt5/include -I/Users/holgermacmini/dev/mxe_qt5152/usr/i686-w64-mingw32.shared.dw2/qt5/include/QtWidgets -I/Users/holgermacmini/dev/mxe_qt5152/usr/i686-w64-mingw32.shared.dw2/qt5/include/QtGui -I/Users/holgermacmini/dev/mxe_qt5152/usr/i686-w64-mingw32.shared.dw2/qt5/include/QtCore/5.15.2 -I/Users/holgermacmini/dev/mxe_qt5152/usr/i686-w64-mingw32.shared.dw2/qt5/include/QtCore/5.15.2/QtCore -I/Users/holgermacmini/dev/mxe_qt5152/usr/i686-w64-mingw32.shared.dw2/qt5/include/QtNetwork -I/Users/holgermacmini/dev/mxe_qt5152/usr/i686-w64-mingw32.shared.dw2/qt5/include/QtCore -I.moc/release -I/Users/holgermacmini/dev/mxe_qt5152/usr/i686-w64-mingw32.shared.dw2/qt5/mkspecs/win32-g++  -o .obj/release/moc_directshowcameraimageencodercontrol.o .moc/release/moc_directshowcameraimageencodercontrol.cpp
i686-w64-mingw32.shared.dw2-g++ -Wl,-s -shared -Wl,-subsystem,windows -Wl,--out-implib,/Users/holgermacmini/dev/mxe_qt5152/tmp-qtmultimedia-i686-w64-mingw32.shared.dw2/qtmultimedia-everywhere-src-5.15.2/plugins/mediaservice/libdsengine.a -o ../../../plugins/mediaservice/dsengine.dll .obj/release/dsserviceplugin.o .obj/release/directshowbasefilter.o .obj/release/directshoweventloop.o .obj/release/directshowmediatype.o .obj/release/directshowmediatypeenum.o .obj/release/directshowpin.o .obj/release/directshowpinenum.o .obj/release/directshowvideobuffer.o .obj/release/directshowutils.o .obj/release/directshowvideoprobecontrol.o .obj/release/directshowaudioprobecontrol.o .obj/release/directshowsamplegrabber.o .obj/release/directshowioreader.o .obj/release/directshowiosource.o .obj/release/directshowplayercontrol.o .obj/release/directshowplayerservice.o .obj/release/directshowvideorenderercontrol.o .obj/release/videosurfacefilter.o .obj/release/directshowaudioendpointcontrol.o .obj/release/directshowmetadatacontrol.o .obj/release/vmr9videowindowcontrol.o .obj/release/evrvideowindowcontrol.o .obj/release/evrcustompresenter.o .obj/release/evrd3dpresentengine.o .obj/release/evrhelpers.o .obj/release/evrdefs.o .obj/release/directshowevrvideowindowcontrol.o .obj/release/dscameraservice.o .obj/release/dscameracontrol.o .obj/release/dsvideorenderer.o .obj/release/dsvideodevicecontrol.o .obj/release/dsimagecapturecontrol.o .obj/release/dscamerasession.o .obj/release/dscameraviewfindersettingscontrol.o .obj/release/dscameraimageprocessingcontrol.o .obj/release/directshowcameraexposurecontrol.o .obj/release/directshowcameracapturedestinationcontrol.o .obj/release/directshowcameracapturebufferformatcontrol.o .obj/release/directshowcamerazoomcontrol.o .obj/release/directshowcameraimageencodercontrol.o .obj/release/moc_dsserviceplugin.o .obj/release/moc_directshoweventloop.o .obj/release/moc_directshowvideoprobecontrol.o .obj/release/moc_directshowaudioprobecontrol.o .obj/release/moc_directshowsamplegrabber.o .obj/release/moc_directshowioreader.o .obj/release/moc_directshowplayercontrol.o .obj/release/moc_directshowplayerservice.o .obj/release/moc_directshowvideorenderercontrol.o .obj/release/moc_videosurfacefilter.o .obj/release/moc_directshowaudioendpointcontrol.o .obj/release/moc_directshowmetadatacontrol.o .obj/release/moc_vmr9videowindowcontrol.o .obj/release/moc_evrvideowindowcontrol.o .obj/release/moc_dscameraservice.o .obj/release/moc_dscameracontrol.o .obj/release/moc_dsvideorenderer.o .obj/release/moc_dsvideodevicecontrol.o .obj/release/moc_dsimagecapturecontrol.o .obj/release/moc_dscamerasession.o .obj/release/moc_dscameraimageprocessingcontrol.o .obj/release/moc_directshowcameraexposurecontrol.o .obj/release/moc_directshowcameracapturedestinationcontrol.o .obj/release/moc_directshowcameracapturebufferformatcontrol.o .obj/release/moc_directshowcamerazoomcontrol.o .obj/release/moc_directshowcameraimageencodercontrol.o  -lgdi32 -lmf -lmfplat -lmfuuid -ld3d9 -ldxva2 -lwinmm -levr /Users/holgermacmini/dev/mxe_qt5152/tmp-qtmultimedia-i686-w64-mingw32.shared.dw2/qtmultimedia-everywhere-src-5.15.2/lib/libQt5Multimedia.a /Users/holgermacmini/dev/mxe_qt5152/usr/i686-w64-mingw32.shared.dw2/qt5/lib/libQt5Widgets.a /Users/holgermacmini/dev/mxe_qt5152/usr/i686-w64-mingw32.shared.dw2/qt5/lib/libQt5Gui.a /Users/holgermacmini/dev/mxe_qt5152/usr/i686-w64-mingw32.shared.dw2/qt5/lib/libQt5Network.a /Users/holgermacmini/dev/mxe_qt5152/usr/i686-w64-mingw32.shared.dw2/qt5/lib/libQt5Core.a -lstrmiids -ldmoguids -luuid -lmsdmo -lole32 -loleaut32 .obj/release/dsengine_resource_res.o -lgdi32 -luser32 -lksuser -lamstrmid
make[3]: *** [sub-plugins-make_first] Error 2
make[2]: *** [sub-src-make_first] Error 2
make[1]: *** [build-only-qtmultimedia_i686-w64-mingw32.shared.dw2] Error 2
real	11m54.537s
user	17m32.717s
sys	3m59.948s
------------------------------------------------------------
```

Here is the [log](https://gist.github.com/aquiles2k/6e6ff0bf322c77f6ce489c9b2916a12e)

Since i dont need multimedia, ill just turn it off for webkit with `-DUSE_QT_MULTIMEDIA=OFF` but still wanted to share the error here.

Thank you and best regards!

When just retrying qtmultimedia a second time i get:

```
[warning]     **gettext-0.20.1**
uname -a
Darwin Holgers-Mac-mini.fritz.box 19.6.0 Darwin Kernel Version 19.6.0: Sun Jul  5 00:43:10 PDT 2020; root:xnu-6153.141.1~9/RELEASE_X86_64 x86_64
git log --pretty=tformat:"%H - %s [%ar] [%d]" -1
2c67738fc8c00ee3e108e3935d6f9317b2375eb8 - Update packages.json & build-matrix.html [3 days ago] [ (HEAD -> master, origin/master, origin/HEAD)]
lsb_release -a 2>/dev/null || sw_vers 2>/dev/null || true
ProductName:	Mac OS X
ProductVersion:	10.15.6
BuildVersion:	19G73
autoconf --version 2>/dev/null | head -1
autoconf (GNU Autoconf) 2.69
automake --version 2>/dev/null | head -1
automake (GNU automake) 1.16.1
gcc --version
Apple clang version 11.0.3 (clang-1103.0.32.62)
Target: x86_64-apple-darwin19.6.0
Thread model: posix
InstalledDir: /Applications/Xcode11.6.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
g++ --version
Apple clang version 11.0.3 (clang-1103.0.32.62)
Target: x86_64-apple-darwin19.6.0
Thread model: posix
InstalledDir: /Applications/Xcode11.6.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
python --version
Python 2.7.16
perl --version 2>&1 | head -3

This is perl 5, version 18, subversion 4 (v5.18.4) built for darwin-thread-multi-2level
(with 2 registered patches, see perl -V for more detail)
rm -rf   '/Users/holgermacmini/dev/mxe_qt5152/tmp-qtmultimedia-i686-w64-mingw32.shared.dw2'
mkdir -p '/Users/holgermacmini/dev/mxe_qt5152/tmp-qtmultimedia-i686-w64-mingw32.shared.dw2'
mkdir -p '/Users/holgermacmini/dev/mxe_qt5152/tmp-qtmultimedia-i686-w64-mingw32.shared.dw2/qtmultimedia-everywhere-src-5.15.2'
mkdir -p '/Users/holgermacmini/dev/mxe_qt5152/tmp-qtmultimedia-i686-w64-mingw32.shared.dw2/qtmultimedia-everywhere-src-5.15.2.build_'
# disable wine with readonly directory
# see https://github.com/mxe/mxe/issues/841
mkdir -p '/Users/holgermacmini/dev/mxe_qt5152/tmp-qtmultimedia-i686-w64-mingw32.shared.dw2/readonly'
chmod 0555 '/Users/holgermacmini/dev/mxe_qt5152/tmp-qtmultimedia-i686-w64-mingw32.shared.dw2/readonly'
cd '/Users/holgermacmini/dev/mxe_qt5152/tmp-qtmultimedia-i686-w64-mingw32.shared.dw2' &&          xz -dc '/Users/holgermacmini/dev/mxe_qt5152/pkg/qtmultimedia-everywhere-src-5.15.2.tar.xz' | tar xf -
cd '/Users/holgermacmini/dev/mxe_qt5152/tmp-qtmultimedia-i686-w64-mingw32.shared.dw2/qtmultimedia-everywhere-src-5.15.2'
cd '/Users/holgermacmini/dev/mxe_qt5152/tmp-qtmultimedia-i686-w64-mingw32.shared.dw2/qtmultimedia-everywhere-src-5.15.2' && '/Users/holgermacmini/dev/mxe_qt5152/usr/i686-w64-mingw32.shared.dw2/qt5/bin/qmake' -after 'LIBS_PRIVATE += -lamstrmid'
Info: creating stash file /Users/holgermacmini/dev/mxe_qt5152/tmp-qtmultimedia-i686-w64-mingw32.shared.dw2/qtmultimedia-everywhere-src-5.15.2/.qmake.stash
Project ERROR: Cannot run target compiler 'i686-w64-mingw32.shared.dw2-g++'. Output:
===================
clang: error: unknown argument: '-fno-keep-inline-dllexport'
Apple clang version 11.0.3 (clang-1103.0.32.62)
Target: x86_64-apple-darwin19.6.0
Thread model: posix
InstalledDir: /Applications/Xcode11.6.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
===================
Maybe you forgot to setup the environment?
make[1]: *** [build-only-qtmultimedia_i686-w64-mingw32.shared.dw2] Error 3

real	0m38.735s
user	0m12.654s
sys	0m2.575s

```

Am i missing something here? Why did it work the first time?

So its failing in sourceresolver.cpp

```
sourceresolver.cpp: In destructor 'virtual SourceResolver::~SourceResolver()':
sourceresolver.cpp:75:25: error: invalid use of incomplete type 'IMFSourceResolver {aka struct IMFSourceResolver}'
         m_sourceResolver->Release();
                         ^
In file included from mfstream.h:44:0,
                 from sourceresolver.cpp:40:
/Users/holgermacmini/dev/mxe_qt5152/usr/i686-w64-mingw32.shared.dw2/include/mfidl.h:3245:19: note: forward declaration of 'IMFSourceResolver {aka struct IMFSourceResolver}'
 typedef interface IMFSourceResolver IMFSourceResolver;
                   ^
```

Seems msys and alexpux came across this too as seen [here](https://gitter.im/msys2/msys2?at=5eb4fd7e3d58de7a3800bb57) on gitter. I'll try to dig further.

--- a/src/qtmultimedia-1.patch
+++ b/src/qtmultimedia-1.patch
@@ -1,0 +1,23 @@
+This file is part of MXE. See LICENSE.md for licensing information.
+
+From 0000000000000000000000000000000000000000 Thu May 27 00:00:00 2021
+From: Adrian Jaekel <aj@elane2k.com>
+Date: Thu, 27 May 2021 14:03:20 +0200
+Subject: [PATCH 1/1] disable wmf plugin for MXE
+
+
+diff --git a/src/plugins/plugins.pro b/src/plugins/plugins.pro
+index 1111111..2222222 100644
+--- a/src/plugins/plugins.pro
++++ b/src/plugins/plugins.pro
+@@ -35,8 +35,9 @@     SUBDIRS += audiocapture \
+                windowsaudio
+
+     qtConfig(directshow): SUBDIRS += directshow
+-    qtConfig(wmf): SUBDIRS += wmf
++#    qtConfig(wmf): SUBDIRS += wmf
++    message(MXE: wmf disabled)
+ }
+ 
+ 
+ winrt {

